tests/test_acasclient.py

```diff
@@ -1376,18 +1376,46 @@ def test_015_get_protocols_by_label(self, experiment):
     @requires_basic_experiment_load
     def test_016_get_experiments_by_protocol_code(self, experiment):
         """Test get experiments by protocol code."""
-        protocols = self.client.get_protocols_by_label("Test Protocol")
         experiments = self.client.\
-            get_experiments_by_protocol_code(protocols[0]["codeName"])
+            get_experiments_by_protocol_code(experiment["protocol"]["codeName"])
         self.assertGreater(len(experiments), 0)
+        # Filter experiments and get the one matching our codeName
+        experiments = [e for e in experiments if e['codeName'] == experiment['codeName']]
+        self.assertEqual(len(experiments), 1)
         self.assertIn('codeName', experiments[0])
         self.assertIn('lsLabels', experiments[0])
+
         self.assertEqual(experiments[0]["lsLabels"][0]["labelText"],
-                         "Test Experiment")
+                         experiment["lsLabels"][0]["labelText"])
```

Comment:
From here up I just changed the test to be a bit more robust because this test is passed a fresh experiment which has a handle on the protocol and experiment code names and labels so we don't need to hardcode this stuff.